sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -214,19 +213,7 @@ def get_all_options(self, drop_default=False):
       subset[str(cls)] = cls
     for cls in subset.values():
       cls._add_argparse_args(parser)  # pylint: disable=protected-access
-    known_args, unknown_args = parser.parse_known_args(self._flags)
-    # Parse args which are not known at this point but might be recognized
-    # at a later point in time, i.e. by the actual Runner.
-    if unknown_args and unknown_args[0] != '':
-      logging.info("Parsing unknown args: %s", unknown_args)
-      for arg in unknown_args:
-        if arg.startswith('--'):
-          parser.add_argument(arg.split('=', 1)[0], nargs='?')
```

Comment:
I believe you need to change here to. Argparse accepts arguments without an equals sign as well. For example `--myargname=myargvalue` would be equivalent to `--myargname myargvalue`. In the latter case, the `split` will return the whole thing.